File: src/main/java/MaximumDepthOfBinaryTree/muzimusi/MaximumDepthOfBinaryTree.java
package MaximumDepthOfBinaryTree.muzimusi;


import sun.reflect.generics.tree.Tree;

// Definition for a binary tree node.
class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;

    TreeNode(int x) {
        val = x;
    }
}

public class MaximumDepthOfBinaryTree {
    public int maxDepth(TreeNode root) {
        if(root==null)
            return 0;
        // 递归调用：左右子树高度最大者+1
        return Math.max(maxDepth(root.left), maxDepth(root.right)) + 1;
    }

    public static void main(String[] args) {
        TreeNode node3 = new TreeNode(3);
        TreeNode node9 = new TreeNode(9);
        TreeNode node20 = new TreeNode(20);
        TreeNode node15 = new TreeNode(15);
        TreeNode node7 = new TreeNode(7);
        node3.left = node9;
        node3.right = node20;
        node20.left = node15;
        node20.right = node7;
        int maxDepth = new MaximumDepthOfBinaryTree().maxDepth(node3);
        System.out.println(maxDepth);
    }
}

File: src/main/java/MaximumDepthOfBinaryTree/GGOBOND/Maximum_Depth_of_Binary_Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def maxDepth(self, root: TreeNode) -> int:
        if root == None:
            return 0

        depth = 0
        queue = [root]
        while len(queue) != 0:  # 遍历根节点的所有子树，如果有子树，则将子树放进列表queue中，然后删除根节点，再将子树作为根节点，循环往复。
            depth += 1
            for i in range(0, len(queue)):
                if queue[0].left:
                    queue.append(queue[0].left)
                if queue[0].right:
                    queue.append(queue[0].right)
                del queue[0]
        return depth